plugin/src/main/java/io/jenkins/plugins/coverage/model/CoverageViewModel.java

```diff
@@ -293,34 +400,65 @@ private SortedMap<CoverageMetric, Coverage> getMetricsDistribution() {
     /**
      * UI table model for the coverage details table.
      */
-    private static class CoverageTableModel extends TableModel {
+    static class CoverageTableModel extends TableModel {
         private final CoverageNode root;
-        private final File buildFolder;
         private final String id;
 
-        CoverageTableModel(final CoverageNode root, final File buildFolder, final String id) {
+        CoverageTableModel(final CoverageNode root, final String id) {
             super();
 
             this.root = root;
-            this.buildFolder = buildFolder;
             this.id = id;
         }
 
         @Override
         public String getId() {
-            return "coverage-details";
+            return id;
+        }
+
+        @Override
+        public TableConfiguration getTableConfiguration() {
+            TableConfiguration tableConfiguration = new TableConfiguration();
+            tableConfiguration.select(SelectStyle.SINGLE);
+            return tableConfiguration;
         }
 
         @Override
         public List<TableColumn> getColumns() {
             List<TableColumn> columns = new ArrayList<>();
 
-            columns.add(new TableColumn("Package", "packageName"));
-            columns.add(new TableColumn("File", "fileName"));
-            columns.add(new TableColumn("Line Coverage", "lineCoverageValue").setHeaderClass(ColumnCss.PERCENTAGE));
-            columns.add(new TableColumn("Line Coverage", "lineCoverageChart", "number"));
-            columns.add(new TableColumn("Branch Coverage", "branchCoverageValue").setHeaderClass(ColumnCss.PERCENTAGE));
-            columns.add(new TableColumn("Branch Coverage", "branchCoverageChart", "number"));
+            // TODO: DataTables API sets options wrong in case of invisible columns (columnDef and <th> number not matching)
+            // TODO: DataTables API requires more granular width options
```

Comment:
Is there anything I can do here in the data tables API?

Reply:
I will show you what I mean in the next meeting.